src/Store.lua

```diff
@@ -144,22 +148,23 @@ function Store:dispatch(action)
 		error("Reducers may not dispatch actions.")
 	end
 
-	local ok, result = pcall(function()
-		self._isDispatching = true
-		self._state = self._reducer(self._state, action)
+	self._isDispatching = true
+	local ok, result = pcall(self._reducer, self._state, action)
+	if ok then
 		self._mutatedSinceFlush = true
-	end)
-
+		self._state = result
+	end
 	self._isDispatching = false
 
 	if not ok then
+		local reducerName = getFunctionName(self._reducer)
 		self._errorReporter.reportReducerError(self._state, action, {
-			message = "Caught error in reducer",
+			message = "Caught error in reducer (" .. reducerName .. ")",
```

Comment:
As far as I can tell, this will only be the name of the very top level reducer. What info does this give us?